core/server/worker/src/main/java/alluxio/worker/block/AsyncCacheRequestManager.java

```diff
@@ -84,23 +88,32 @@ public void submitRequest(Protocol.AsyncCacheRequest request) {
             } catch (BlockDoesNotExistException e) {
               LOG.error("Failed to unlock block on async caching. We should never reach here", e);
             }
+            ASYNC_CACHE_DUPLICATE_REQUESTS.inc();
             return;
           }
           Protocol.OpenUfsBlockOptions openUfsBlockOptions = request.getOpenUfsBlockOptions();
           boolean isSourceLocal = mLocalWorkerHostname.equals(request.getSourceHost());
           // Depends on the request, cache the target block from different sources
           boolean result;
           if (isSourceLocal) {
+            ASYNC_CACHE_UFS_BLOCKS.inc();
             result = cacheBlockFromUfs(blockId, blockLength, openUfsBlockOptions);
           } else {
+            ASYNC_CACHE_REMOTE_BLOCKS.inc();
             InetSocketAddress sourceAddress =
                 new InetSocketAddress(request.getSourceHost(), request.getSourcePort());
             result = cacheBlockFromRemoteWorker(
                     blockId, blockLength, sourceAddress, openUfsBlockOptions);
           }
+          if (result) {
```

Comment:
is it easier to reason to put this block in `finally`?

Reply:
Good point, we just need to change the scope of `result`.